loader/src/platform/android/LoaderImpl.cpp

```diff
@@ -44,3 +44,49 @@ bool Loader::Impl::supportsLaunchArguments() const {
 std::string Loader::Impl::getLaunchCommand() const {
     return std::string(); // Empty
 }
+
+bool Loader::Impl::getPermissionStatus(std::string_view const name) const {
+    JniMethodInfo info;
+    if (JniHelper::getStaticMethodInfo(info, "com/geode/launcher/utils/GeodeUtils", "getPermissionStatus", "(Ljava/lang/String;)Z")) {
+        jstring permString = info.env->NewStringUTF(std::string(name).c_str());
+        jboolean result = info.env->CallStaticBooleanMethod(info.classID, info.methodID, permString);
+        info.env->DeleteLocalRef(info.classID);
+        info.env->DeleteLocalRef(permString);
+
+        return result == JNI_TRUE;
+    }
+
+    return false;
```

Comment:
when you're using new methods from the jni, keep in mind that this isn't enough to prevent a crash if the method isn't found. you can see the clear exception code (which you should add) above in `getGameVersion` (the function is also provided in [utils](https://github.com/geode-sdk/geode/blob/a8ab0e4ba69704a1d9859f265a646cefa4cd3aae/loader/src/platform/android/util.cpp#L46), but it's anonymous so you'll have to copy the contents or make a new function here)